crates/sage-database/src/primitives/cats.rs

```diff
@@ -1,12 +1,26 @@
 use chia::{protocol::Bytes32, puzzles::LineageProof};
 use chia_wallet_sdk::driver::Cat;
-use sqlx::SqliteExecutor;
+use sqlx::{Row, SqliteExecutor};
 
 use crate::{
     into_row, to_bytes, to_bytes32, CatCoinRow, CatCoinSql, CatRow, CatSql, CoinStateRow,
     CoinStateSql, Database, DatabaseTx, FullCatCoinSql, Result,
 };
 
+#[derive(Debug, Clone, Copy)]
+pub enum CoinSortMode {
+    ParentCoinId,
+    Amount,
```

Comment:
Should we have an index on coin amount? Or is it not worth the insert overhead and additional disk space?